### PR TITLE
Resize op

### PR DESCRIPTION
**Pull request by scxiao**

### CODE REVIEW

File: src/onnx/onnx.cpp [hunk 1]
```diff
@@ -1065,6 +1068,184 @@ struct onnx_parser
...
+             }},
+            {"asymmetric",
+             [=](std::size_t, std::size_t, std::size_t idx, double scale) { return idx / scale; }},
+            {"tf_half_pixel_for_nn", [=](std::size_t, std::size_t, std::size_t idx, double scale) {
+                 return (idx + 0.5) / scale;
+             }}};
+
+        if(!contains(idx_ops, mode))
+        {
+            MIGRAPHX_THROW("PRASE_RESIZE: coordinate_transformation_mode " + mode +
```
Comment (kahmed10) [inline src/onnx/onnx.cpp:1125]: typo on PARSE here

**Review by kahmed10 — APPROVED** [review]

**Comment by scxiao** [comment]
@pfultz2 Can you review this PR? Thanks

File: src/onnx/onnx.cpp [hunk 2]
```diff
@@ -1067,6 +1070,184 @@ struct onnx_parser
         return prog.add_instruction(op, make_contiguous(args[0]));
     }
 
+    const nearest_op& get_nearest_op(const std::string& mode)
```
Comment (pfultz2) [inline src/onnx/onnx.cpp:1073]: Use `auto&` for return.

File: src/onnx/onnx.cpp [hunk 3]
```diff
@@ -52,7 +52,9 @@ namespace onnx = onnx_for_migraphx;
 
 struct onnx_parser
 {
-    using attribute_map = std::unordered_map<std::string, onnx::AttributeProto>;
+    using attribute_map   = std::unordered_map<std::string, onnx::AttributeProto>;
+    using nearest_op      = std::function<std::size_t(std::size_t, double)>;
+    using original_idx_op = std::function<double(std::size_t, std::size_t, std::size_t, double)>;
```
Comment (pfultz2) [inline src/onnx/onnx.cpp:57]: Can these typedefs be moved into each corresponding function?

File: src/onnx/onnx.cpp (hunk 2 above)
Comment (scxiao) [reply src/onnx/onnx.cpp]: fixed

File: src/onnx/onnx.cpp (hunk 3 above)
Comment (scxiao) [reply src/onnx/onnx.cpp]: fixed

File: src/onnx/onnx.cpp [hunk 4]
```diff
@@ -1067,6 +1068,187 @@ struct onnx_parser
         return prog.add_instruction(op, make_contiguous(args[0]));
     }
 
+    const auto& get_nearest_op(const std::string& mode)
```
Comment (pfultz2) [inline src/onnx/onnx.cpp:1071]: These functions can be declared `static`, I believe? I dont think they need to be a member function.

Comment (scxiao) [reply src/onnx/onnx.cpp]: fixed

**Review by pfultz2 — APPROVED** [review]

**Pull request merged**